vdr/didservice/resolvers.go

```diff
@@ -164,16 +168,41 @@ func (r KeyResolver) ResolveRelationKey(keyID string, validAt *time.Time, relati
 		ResolveTime: validAt,
 	})
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	relationships, err := resolveRelationships(doc, relationType)
+	if err != nil {
+		return nil, err
 	}
-	relationships, _ := resolveRelationships(doc, relationType)
-
 	for _, rel := range relationships {
 		if rel.ID.String() == keyID {
 			return rel.PublicKey()
 		}
 	}
-	return "", types.ErrKeyNotFound
+	return nil, types.ErrKeyNotFound
+}
+
+func (r KeyResolver) ResolveKey(id did.DID, validAt *time.Time, relationType types.RelationType) (ssi.URI, crypto.PublicKey, error) {
+	doc, _, err := r.Store.Resolve(id, &types.ResolveMetadata{
+		ResolveTime: validAt,
+	})
+	if err != nil {
+		return ssi.URI{}, nil, err
+	}
+	keys, err := resolveRelationships(doc, relationType)
+	if err != nil {
+		return ssi.URI{}, nil, err
+	}
+	for _, key := range keys {
+		kid := key.ID.String()
+		u, _ := ssi.ParseURI(kid)
```

Comment:
Key IDs are unmarshalled as `did.DID` (should probably have been ssi.URI?), so it can't fail (since it's already a valid URI). Now I think of it, we can just call `URI()` on the ID, not having to parse it again...